core/include/kernel/tee_ta_manager.h

```diff
@@ -164,11 +167,16 @@ struct tee_ta_session *tee_ta_get_session(uint32_t id, bool exclusive,
 void tee_ta_put_session(struct tee_ta_session *sess);
 
 #if defined(CFG_TA_GPROF_SUPPORT) || defined(CFG_TA_FTRACE_SUPPORT)
-void tee_ta_update_session_utime_suspend(void);
-void tee_ta_update_session_utime_resume(void);
+void tee_ta_update_session_utime_suspend(uint32_t flag);
+void tee_ta_update_session_utime_resume(uint32_t flag);
```

Comment:
With the introduction of this `flag` parameter, and the fact that `FTRACE_UPDATE_FLAG` does not account only for time in user space, I think the name of the function has become unclear. Same for `tee_ta_update_session_utime(bool suspend, uint32_t flag)` below.
It would be better to not try and re-use those, and add functions specific to the ftrace needs.

Reply:
Okay, will add new APIs.